src/emulator/download.js

```diff
@@ -35,9 +36,24 @@ module.exports = (name) => {
           )
         );
       } else {
-        fs.copySync(tmpFile.name, emulator.localPath);
-        fs.chmodSync(emulator.localPath, 0o755);
-        resolve();
+        const hash = crypto.createHash('md5'),
+        const stream = fs.createReadStream(tmpFile.name);
+        stream.on('data', data => hash.update(data));
+        stream.on('end', function() {
+          const checksum = hash.digest('hex');
+          if (checksum != emulator.expectedHash) {
+            reject(
+              new FirebaseError(
+                `download failed, expected checksum ${emulator.expectedHash} but got ${checksum}`,
+                { exit: 1 }
+              )
+            );
+          } else {
```

Comment:
this else block can be removed if the `reject` above is returned

Reply:
(and the same is true of the `else` block that _this_ entire check is in)

Reply:
converted these to ternary statements, lmk if that's non-idiomatic